jupyterlab_gcpscheduler/src/components/dialog.tsx

```diff
@@ -88,6 +84,7 @@ interface Props {
 interface State {
   dialogClosedByUser: boolean;
   gcpSettings?: GcpSettings;
+  metadataProjectId?: string;
```

Comment:
Correct me if I am wrong, but I believe you introducing this because we will no longer have the initialization component that was responsible for setting the projectId in settings? 

Do we need to have this additional state variable if we would just move the logic that retrieves projectId and sets it to this module? Where would projectId ultimately be set in the GcpSettings?

Reply:
So for me, gcpSettings was cleared so I noticed this was necessary. I thought it was possible other users may still have it and it could be a good fallback

Reply:
I guess my question is around where you would expect the `projectId` setting to be saved? Since we used to do it in the Initialization component that is no longer present, I think it would be best if somehow the logic inside of the `_settingsChanged` method would be to check to see if `projectId` was set, and if not, retrieve it from the `gcpService` property.

In the current implementation, it looks as if you'd be retrieving it from the server each time but never actually storing it in settings. That appears to be why you have to pass `projectId` into SchedulerForm explicitly rather than rely on the setting property to be present.

Reply:
Ah I see what you mean. initially I thought we could just phase out the gcpSettings.projectId, but I can use it.

Reply:
Where do you think the gcpSettings.projectId should be set then?

Reply:
It's your call. The original intent of `GcpSettings.projectId` was to facilitate the idea of using the extension _outside_ a GCE VM, but that's no longer a goal.

The only disadvantage I see is that we will always be introducing an extra call to the backend when the extension is added to the DOM.